webapp/src/components/KeyboardControlsHelp.tsx

```diff
@@ -0,0 +1,177 @@
+                    <Row name={lf("Move mode: Cancel")} shortcuts={[ShortcutNames.EXIT]} />
+                    <Row name={lf("Disconnect blocks")} shortcuts={[ShortcutNames.DISCONNECT]} />
+                </tbody>
+            </table>
+        </aside>
+    );
+}
+
+const Shortcut = ({ keys }: { keys: string[] }) => {
+    const joiner = pxt.BrowserUtils.isMac() ? " " : " + "
+    return (
+        <span className="shortcut">
+            {keys.reduce((acc, key) => {
+                return acc.length === 0
+                    ? [...acc,  <Key key={key} value={key} />]
+                    : [...acc, joiner, <Key key={key} value={key} />]
+            }, [])}
+        </span>
+    );
+}
+
+interface RowProps {
+    name: string;
+    shortcuts?: Array<string | string[]>;
+    joiner?: string;
+    children?: React.ReactNode;
+}
+
+const Row = ({ name, shortcuts = [], joiner, children}: RowProps) => {
+    const shortcutElements = shortcuts.map((s, idx) => {
+        if (typeof s === "string") {
+            // Pull keys from shortcut registry.
+            return <Shortcut key={idx} keys={getActionShortcut(s)} />
+        } else {
+            // Display keys as specified.
+            return <Shortcut key={idx} keys={s} />
+        }
+    })
+    return joiner === "row" ? (
+        <>
+            <tr>
+                <td width="50%" rowSpan={shortcuts.length}>{name}</td>
+                <td width="50%">
+                    {shortcutElements[0]}
+                </td>
+            </tr>
+            {shortcutElements.map((el, idx) => idx === 0
+                ? undefined
+                : (<tr key={idx}>
+                        <td width="50%">
+                            {el}
+                        </td>
+                    </tr>))}
+        </>
+    ) : (
+        <tr>
+            <td width="50%">{name}</td>
+            <td width="50%">
+                {shortcutElements.reduce((acc, shortcut) => {
+                    return acc.length === 0
+                        ? [...acc,  shortcut]
+                        : [...acc, joiner ? ` ${joiner} ` : " ", shortcut]
+                }, [])}
+                {children}
+                <br />
+            </td>
+        </tr>
+    )
+}
+
+const Key = ({ value }: { value: string }) => {
+    let aria;
+    switch (value) {
+        case "↑": {
+            aria = lf("Up Arrow");
+            break;
+        }
+        case "↓": {
+            aria = lf("Down Arrow");
+            break;
+        }
+        case "←": {
+            aria = lf("Left Arrow");
+            break;
+        }
+        case "→": {
+            aria = lf("Right Arrow");
+            break;
+        }
+        case "⌘": {
+            aria = lf("Command");
+            break;
+        }
+        case "⌥": {
+            aria = lf("Option");
+            break;
+        }
+    }
+    return <span className="key" aria-label={aria}>{lf("{0}", value)}</span>
```

Comment:
no need to use `lf()` here since it's used above!

Reply:
Thanks. This turned out to be more interesting than expected. There were a mix: shortcuts that were in some way hardcoded had already been through `lf` but ones from the registry where not otherwise going through `lf`. I suppose even though `lf` was being called via the line above for the registry ones, the strings would likely not be extracted correctly.

I have restructured to do the translation more directly from the shortcut registry format to translated text in [51b63a2](https://github.com/microsoft/pxt/pull/10556/commits/51b63a296af39827808830c11dcd48f09ab83956). I also managed to drop the hardcoding for the shortcuts where we document two alternatives. Follow up to fix `lf` quote style in 0183179156785a6db79d0c25846b6ca837dd61f5.